src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Sources/ManualResetValueTaskSourceCore.cs

```diff
@@ -117,28 +119,34 @@ private void ThrowForFailedGetResult(short token)
         /// <param name="flags">The flags describing the behavior of the continuation.</param>
         public void OnCompleted(Action<object?> continuation, object? state, short token, ValueTaskSourceOnCompletedFlags flags)
         {
-            ArgumentNullException.ThrowIfNull(continuation);
-
+            if (continuation is null)
+            {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.continuation);
```

Comment:
This is because of the generic context, right?

Should we add a comment to such helpers so future refactorings don't accidentally regress things?

Reply:
We use these ThrowHelpers throughout corelib in literally thousands of places.  I wouldn't want to comment each usage as to why it's being used.  And the ThrowHelper class itself has a large comment about how it's there to reduce code size.

Reply:
I meant specifically this vs `ArgumentNullException.ThrowIfNull(continuation)`

Reply:
Sure, but that goes for 456 occurrences of ThrowHelper.ThrowArgumentNullException(ExceptionArgument.whatever) elsewhere, too.